This gives a failure like:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/jason/src/elastic/elasticsearch/qa/wildfly/build.gradle' line: 39

* What went wrong:
A problem occurred evaluating project ':qa:wildfly'.
> repository on project [wildfly] is using http for artifacts on [http://download.jboss.org]

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 12s
```

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -582,7 +582,7 @@ class BuildPlugin implements Plugin<Project> {
 
     /** Adds repositories used by ES dependencies */
     static void configureRepositories(Project project) {
-        project.getRepositories().whenObjectAdded { repository ->
+        project.getRepositories().all { repository ->
             if (repository instanceof MavenArtifactRepository) {
                 final MavenArtifactRepository maven = (MavenArtifactRepository) repository
                 assertRepositoryURIUsesHttps(project, maven.getUrl())
@@ -631,7 +631,7 @@ class BuildPlugin implements Plugin<Project> {
 
     private static void assertRepositoryURIUsesHttps(final Project project, final URI uri) {
         if (uri != null && uri.toURL().getProtocol().equals("http")) {
-            throw new GradleException("repository on project [" + project.name + "] is using http for artifacts on [" + uri.toURL() + "]")
+            throw new GradleException("repository on project with path [${project.path}] is using http for artifacts on [${uri.toURL()}]")
         }
     }
 

